### PR TITLE
AddRef: update return type based on base class

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: Compositor/test/compositortest/Test.cpp [hunk 1]
```diff
@@ -44,8 +44,9 @@ class Keyboard : public Compositor::IDisplay::IKeyboard {
     ~Keyboard() override = default;
 
 public:
-    virtual void AddRef() const override
+    virtual uint32_t AddRef() const override
     {
+        return 0;
```
Comment (pwielders) [inline Compositor/test/compositortest/Test.cpp:49]: return(Core::ERROR_COMPOSIT_OBJECT);

File: Streamer/Frontend.h [hunk 2]
```diff
@@ -263,9 +263,9 @@ namespace Player {
             }
 
         public:
-            void AddRef() const override
+            uint32_t AddRef() const override
             {
-                Core::InterlockedIncrement(_refCount);
+                return (Core::InterlockedIncrement(_refCount));
             }
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply Streamer/Frontend.h:271]: done

File: Compositor/test/compositortest/Test.cpp (hunk 1 above)
Comment (HaseenaSainul) [reply Compositor/test/compositortest/Test.cpp]: done

File: Compositor/lib/Mesa/renderer/test/texturebuffer/TextureBuffer.cpp [hunk 3]
```diff
@@ -4,8 +4,9 @@
 
 namespace WPEFramework {
 namespace Compositor {
-    void TextureBuffer::AddRef() const
+    uint32_t TextureBuffer::AddRef() const
     {
+        return 0;
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply Compositor/lib/Mesa/renderer/test/texturebuffer/TextureBuffer.cpp]: done

File: Compositor/lib/Mesa/Compositor.cpp [hunk 4]
```diff
@@ -308,9 +308,10 @@ namespace Plugin {
              * Exchange::ICompositionBuffer methods
              */
 
-            void AddRef() const override
+            uint32_t AddRef() const override
             {
                 // We are not a reference counted object, so do nothing.
+                return (0);
```
[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply Compositor/lib/Mesa/Compositor.cpp]: done

**Comment by HaseenaSainul** [comment]
> <I think we are missing still some "return(Core::ERROR_COMPOSIT_OBJECT);" where you currently return 0. See comments.

Sorry missed this comments

File: Compositor/lib/Mesa/renderer/test/texturebuffer/TextureBuffer.cpp [hunk 5]
```diff
@@ -4,13 +4,14 @@
...
-    void TextureBuffer::AddRef() const
+    uint32_t TextureBuffer::AddRef() const
     {
+        return (Core::ERROR_COMPOSIT_OBJECT);
     }
 
     uint32_t TextureBuffer::Release() const
     {
-        return 0;
+        return (Core::ERROR_COMPOSIT_OBJECT);
```
Comment (pwielders) [inline Compositor/lib/Mesa/renderer/test/texturebuffer/TextureBuffer.cpp:14]: return (Core::ERROR_NONE);

File: Compositor/lib/Mesa/Compositor.cpp [hunk 6]
```diff
@@ -308,14 +308,15 @@ namespace Plugin {
...
+            uint32_t AddRef() const override
             {
                 // We are not a reference counted object, so do nothing.
+                return (Core::ERROR_COMPOSIT_OBJECT);
             }
             uint32_t Release() const override
             {
                 // We are not a reference counted object, so do nothing.
-                return (0);
+                return (Core::ERROR_COMPOSIT_OBJECT);
```
Comment (pwielders) [inline Compositor/lib/Mesa/Compositor.cpp:319]: return (Core::ERROR_NONE);

**Review by pwielders — COMMENTED** [review]
Sorry! Did not pay enough attention here. The Release (where you have return 0) can always return (Core::ERROR_NONE);

File: Compositor/test/compositortest/Test.cpp [hunk 7]
```diff
@@ -44,12 +44,13 @@ class Keyboard : public Compositor::IDisplay::IKeyboard {
...
 
 public:
-    virtual void AddRef() const override
+    virtual uint32_t AddRef() const override
     {
+        return (Core::ERROR_COMPOSIT_OBJECT);
     }
     virtual uint32_t Release() const override
     {
-        return 0;
```
Comment (pwielders) [inline Compositor/test/compositortest/Test.cpp:52]: return (Core::ERROR_NONE);

**Comment by pwielders** [comment]
After giving it a second thought, I think your changes are actually very good. Lets also return Core::ERROR_COMPOSIT_OBJECT on the Release of an AddRef() that return (Core::ERROR_COMPOSIT_OBJECT) so that it is symetrical (I love symetrics :-) )
But it does mean that we also have to do so on Core::ServiceAdministrator and Core::SinkType in core/Services.h. Could you change it there as well and ignore my comments here :-)

**Comment by HaseenaSainul** [comment]
> Sorry! Did not pay enough attention here. The Release (where you have return 0) can always return (Core::ERROR_NONE);

As we discussed, since it also not doing InterlockedDecrement, I would like to keep ERROR_COMPOSIT_OBJECT for all Release where it return 0

**Comment by pwielders** [comment]
I think we have crossing messages :-) As stated just before: You are right!

**Comment by HaseenaSainul** [comment]
> After giving it a second thought, I think your changes are actually very good. Lets also return Core::ERROR_COMPOSIT_OBJECT on the Release of an AddRef() that return (Core::ERROR_COMPOSIT_OBJECT) so that it is symetrical (I love symetrics :-) ) But it does mean that we also have to do so on Core::ServiceAdministrator and Core::SinkType in core/Services.h. Could you change it there as well and ignore my comments here :-)

Sure, I will do

**Review by pwielders — APPROVED** [review]

**Pull request merged**